### PR TITLE
have different Config implementations for different channels

### PR DESCRIPTION
**Pull request by amircodota**

### CODE REVIEW

**Comment by boaz-codota** [comment]
@amircodota I have no idea what this suppose to do

**Comment by amircodota** [comment]
@boaz-codota Ths idea is to have a Config class with constants for each of the supported channels - alpha, beta, production, and include the right Config.java based on the given channel.
This way, we can have a different configuration based on the channel we are compiling for, so for example, we can know which channel we are on, what url to download the binary from etc.

**Comment by boaz-codota** [comment]
> @boaz-codota Ths idea is to have a Config class with constants for each of the supported channels - alpha, beta, production, and include the right Config.java based on the given channel.
> This way, we can have a different configuration based on the channel we are compiling for, so for example, we can know which channel we are on, what url to download the binary from etc.

Is it a prep for the future? (If so, please note that "premature optimisation is the root of all evil")

**Comment by amircodota** [comment]
@boaz-codota hehe... no, I use it to pass the clientChannel metadata field to the binary

File: src/main/java/com/tabnine/binary/BinaryRun.java [hunk 1]
```diff
@@ -28,7 +29,8 @@ public BinaryRun(BinaryVersionFetcher binaryFetcher) {
             List<String> metadata = new ArrayList<>(asList(
                     "--client-metadata",
                     "pluginVersion=" + Utils.getPluginVersion(),
-                    "clientIsUltimate=" + PlatformUtils.isIdeaUltimate()
+                    "clientIsUltimate=" + PlatformUtils.isIdeaUltimate(),
+                    "clientChannel=" + Config.CHANNEL
```
Comment (boaz-codota) [inline src/main/java/com/tabnine/binary/BinaryRun.java:33]: This is high risk change. The binary won't run if he does not recognise that parameter

**Review by boaz-codota — APPROVED** [review]

**Pull request merged**